ballerina/tests/error.bal

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'string as strings;
+import ballerina/test;
+import ballerina/sql;
+
+string errorDB = "ERROR_DB";
+
+@test:Config {
+    groups: ["error"]
+}
+function TestAuthenticationError() {
+    Client|error dbClient = new (host, "user", password, errorDB, port);
+    test:assertTrue(dbClient is sql:ApplicationError);
+    sql:ApplicationError sqlerror = <sql:ApplicationError> dbClient;
+    test:assertTrue(strings:includes(sqlerror.message(), "Access denied for user"), sqlerror.message());
+}
+
+@test:Config {
+    groups: ["error"]
+}
+function TestLinkFailure() {
+    Client|error dbClient = new ("host", user, password, errorDB, port);
```

Comment:
```suggestion
    Client|error error = new ("host", user, password, errorDB, port);
```

Better to rename as error, since we are expecting one here.